app/controllers/organizations/pets_controller.rb

```diff
@@ -29,11 +29,21 @@ def show
   end
 
   def create
+    transaction_success = false
     @pet = Pet.new(pet_params)
 
-    if @pet.save
+    ActiveRecord::Base.transaction do
+      if @pet.save && Organizations::DefaultPetTaskService.new(@pet).create_tasks
+        transaction_success = true
+      else
+        raise ActiveRecord::Rollback
+      end
+    end
+
+    if transaction_success
```

Comment:
If you use the approach I suggest above you can do `@pet.persisted?` instead to know if the transaction worked.